### PR TITLE
Ruff: enable more code quality rules

### PR DESCRIPTION
**Pull request by jnewb1**
With ruff, some of the rules must be explicity enabled, even ones that were enabled by default in pylint. Is there any other rules that we want?

https://beta.ruff.rs/docs/rules/#flake8-implicit-str-concat-isc

https://github.com/commaai/openpilot/pull/29299

@sshane

### CODE REVIEW

File: selfdrive/car/docs_definitions.py [hunk 1]
```diff
@@ -174,12 +174,12 @@ def all_parts(self):
...
   EXP_LONG_AVAIL = CarFootnote(
-    "Experimental openpilot longitudinal control is available behind a toggle; " +
+    "Experimental openpilot longitudinal control is available behind a toggle; "
     "the toggle is only available in non-release branches such as `devel` or `master-ci`. ",
     Column.LONGITUDINAL, docs_only=True)
   EXP_LONG_DSU = CarFootnote(
-    "By default, this car will use the stock Adaptive Cruise Control (ACC) for longitudinal control. " +
-    "If the Driver Support Unit (DSU) is disconnected, openpilot ACC will replace " +
+    "By default, this car will use the stock Adaptive Cruise Control (ACC) for longitudinal control. "
+    "If the Driver Support Unit (DSU) is disconnected, openpilot ACC will replace "
```
Comment (sshane) [inline selfdrive/car/docs_definitions.py:182]: We don't want implicit concatenation

**Comment by adeebshihadeh** [comment]
Let's try these:
* https://beta.ruff.rs/docs/rules/#flake8-builtins-a
* https://beta.ruff.rs/docs/rules/#flake8-comprehensions-c4
* https://beta.ruff.rs/docs/rules/#flake8-pie-pie


Any changes we make should also be propagated to the submodule configs too.

**Comment by sshane** [comment]
Also noticed we don't catch mutable args in functions too (`dangerous-default-value`). Let's make sure we set up ruff to catch the most common bugs to write.

**Comment by jnewb1** [comment]
dangerous-default-value is B006, from flake8-bugbear. flake8-implicit-str-concat.allow-multiline=false disabled implicit concat across multiple lines.

However, ISC003 disables explicit string concat
```
    "Experimental openpilot longitudinal control is available behind a toggle; " +
    "the toggle is only available in non-release branches such as `devel` or `master-ci`. ",
```

in favor of implicit with parens
```
 ("Experimental openpilot longitudinal control is available behind a toggle; "
    "the toggle is only available in non-release branches such as `devel` or `master-ci`. ")
```

So I disabled ISC003, unless we like the parens better?

**Comment by jnewb1** [comment]
splitting into multiple PRS

**Pull request closed without merging**